packages/core/src/types/spring.ts

```diff
@@ -55,12 +55,22 @@ export type SpringUpdate<T extends object = {}> = Partial<T> &
  * The `T` parameter should only contain animated props.
  */
 export interface SpringUpdateFn<T extends object = {}> {
+  /** Update the props of a spring */
+  (props: SpringUpdate<T>): void
+}
+
+/**
+ * Imperative API for the useSprings hook, allowing to update individually or over the array
+ *
+ * The `T` parameter should only contain animated props.
+ */
+export interface SpringsUpdateFn<T extends object = {}> {
   /** Update the props of a spring */
   (
     props:
       | SpringUpdate<T>
       | SpringUpdate<T>[]
-      | ((index: number, spring: Controller<T>) => SpringUpdate<T>)
+      | ((index: number, spring?: Controller<T>) => SpringUpdate<T>)
```

Comment:
This is wrong, because the controller is always passed.

Reply:
ah, my setter call didn't need a second arg, but I guess you're defaulting it?

Reply:
Sorry, not sure what you mean. The `?` type operator only affects the caller. The callback is allowed to omit any trailing arguments it doesn't need.

```ts
const [props, set] = useSprings(i => { ... })

// Both of these calls are valid in TypeScript's eyes
set(i => { ... })
set((i, spring) => { ... })
```

Reply:
ah of course - thanks!